### PR TITLE
feat(sdk): fix client tls connections

### PR DESCRIPTION
**Pull request by QuantumExplorer**
## Issue being fixed or feature implemented
Tls connections for clients were no longer working, this now fixes things.

## What was done?
Added tls explicit configuration.

## How Has This Been Tested?
Verified it was working

## Breaking Changes
Fixes issues.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [x] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [X] I have assigned this pull request to a milestone

## Summary by CodeRabbit

- **New Features**
  - Enhanced error handling for the transport client, providing more contextual information during failures.
  - Improved error propagation when creating `PlatformGrpcClient`.

- **Bug Fixes**
  - Refined error reporting for transport errors, offering better specificity.

- **Refactor**
  - Updated method signatures to return results for improved error management.
  - Modified the `get_or_create` method to support error handling during item creation.

- **Chores**
  - Adjusted dependency configuration for better integration with `dapi-grpc`, now including specific features.

### CODE REVIEW

File: packages/rs-dapi-client/src/transport/grpc.rs [hunk 1]
```diff
@@ -18,59 +18,101 @@ pub type PlatformGrpcClient = PlatformClient<Channel>;
...
-    let mut builder = Channel::builder(uri);
+fn create_channel(
+    uri: Uri,
+    settings: Option<&AppliedRequestSettings>,
+) -> Result<Channel, dapi_grpc::tonic::transport::Error> {
+    let mut builder = Channel::builder(uri).tls_config(
+        ClientTlsConfig::new()
+            .with_native_roots()
+            .with_webpki_roots()
+            .assume_http2(true),
```
Comment (lklimek) [inline packages/rs-dapi-client/src/transport/grpc.rs:29]: I think we should read these details from settings.

File: packages/rs-dapi-client/src/connection_pool.rs [hunk 2]
```diff
@@ -67,19 +67,21 @@ impl ConnectionPool {
     /// * `prefix` -  Prefix for the item in the pool. Used to distinguish between Core and Platform clients.
     /// * `uri` - URI of the node.
     /// * `settings` - Applied request settings.
-    pub fn get_or_create(
+    pub fn get_or_create<T>(
```
Comment (lklimek) [inline packages/rs-dapi-client/src/connection_pool.rs:70]: T is confusing, if it's error maybe name it E, or add some constraint?

**Review by shumkov — COMMENTED** [review]
Tests aren't passing

File: packages/rs-dapi-client/src/transport/grpc.rs (hunk 1 above)
Comment (QuantumExplorer) [reply packages/rs-dapi-client/src/transport/grpc.rs:29]: Well... this is the same behavior I believe as before, just now it needs to be explicit.

**Review by lklimek — APPROVED** [review]

**Pull request merged**